components/esp_diag_data_store/src/rtc_store/rtc_store.c

```diff
@@ -556,11 +556,12 @@ static esp_err_t rtc_store_meta_hdr_init()
 
     // populate meta header
 #if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
-    esp_app_get_elf_sha256(s_priv_data.sha_sum, sizeof(s_priv_data.sha_sum));
+    const uint8_t* src = esp_app_get_description()->app_elf_sha256;
+    memcpy((uint8_t *)s_priv_data.meta_hdr->sha_sum, src, RTC_STORE_SHA_SIZE);
 #else
     esp_ota_get_app_elf_sha256(s_priv_data.sha_sum, sizeof(s_priv_data.sha_sum));
+    hex_to_bytes((uint8_t *) s_priv_data.sha_sum, (uint8_t *) s_priv_data.meta_hdr->sha_sum, RTC_STORE_SHA_SIZE);
```

Comment:
Can you please wrap `hex_to_bytes` definition under `#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)` check?
It will emit unused static function error otherwise for newer IDFs when strict compiler flags are set.